app/screens/reactions/emoji_bar/index.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useRef} from 'react';
+import {StyleSheet} from 'react-native';
+import {FlatList} from 'react-native-gesture-handler';
+
+import Item from './item';
+
+import type ReactionModel from '@typings/database/models/servers/reaction';
+
+type Props = {
+    emojiSelected: string;
+    reactionsByName: Map<string, ReactionModel[]>;
+    setIndex: (idx: number) => void;
+    sortedReactions: string[];
+}
+
+type ScrollIndexFailed = {
+    index: number;
+    highestMeasuredFrameIndex: number;
+    averageItemLength: number;
+};
+
+const style = StyleSheet.create({
+    container: {
+        maxHeight: 44,
+    },
+});
+
+const EmojiBar = ({emojiSelected, reactionsByName, setIndex, sortedReactions}: Props) => {
+    const listRef = useRef<FlatList<string>>(null);
+
+    const scrollToIndex = (index: number, animated = false) => {
+        listRef.current?.scrollToIndex({
+            animated,
+            index,
+            viewOffset: 0,
+            viewPosition: 1, // 0 is at bottom
+        });
+    };
+
+    const onPress = useCallback((emoji: string) => {
+        const index = sortedReactions.indexOf(emoji);
+        setIndex(index);
+    }, [sortedReactions]);
+
+    const onScrollToIndexFailed = useCallback((info: ScrollIndexFailed) => {
+        const index = Math.min(info.highestMeasuredFrameIndex, info.index);
+
+        scrollToIndex(index);
+    }, []);
+
+    const renderItem = useCallback(({item}) => {
+        return (
+            <Item
+                count={reactionsByName.get(item)?.length || 0}
+                emojiName={item}
+                highlight={item === emojiSelected}
+                onPress={onPress}
+            />
+        );
+    }, [sortedReactions, emojiSelected, reactionsByName]);
```

Comment:
Just for future proof, depend on `onPress` instead of on `sortedReactions`.

Reply:
I made the change.

Also we should start soon to run some profiling to see which `useMemo` `useCallback` can be removed, I fear we are adding a bunch of them at that may be more expensive than letting certain components re-render